src/crypto_helper/computations.rs

```diff
@@ -92,3 +94,38 @@ pub fn process_zlib(input: &ZlibInput) -> Result<Vec<u8>, String> {
         }
     }
 }
+
+pub fn process_argon2(input: &Argon2Input) -> Result<Vec<u8>, String> {
+    match &input.action {
+        Argon2Action::Hash(hash_action) => {
+            let argon2ctx = argon2::Argon2::new(
+                hash_action.variant.into(),
+                hash_action.version.into(),
+                hash_action.into(),
+            );
+            let b64 = base64::encode(&hash_action.salt);
+            let salt = argon2::password_hash::Salt::from_b64(&b64).unwrap();
```

Comment:
1. add the salt generation of the `hash_action.salt` is empty
2. handle errors here properly. too short salt will produce an error here

```suggestion
            let b64 = base64::encode(&hash_action.salt); // handle an empty salt here
            let salt = argon2::password_hash::Salt::from_b64(&b64)
                .map_err(|err| format!("Error while building a salt: {:?}", err))?;
```